main/lsp/LSPFileUpdates.h

```diff
@@ -47,6 +49,35 @@ class LSPFileUpdates final {
      * Returns a copy of this LSPFileUpdates object. Does not handle deepCopying `updatedGS`.
      */
     LSPFileUpdates copy() const;
+
+    struct FastPathFilesToTypecheckResult {
+        // size_t is an index into the updatedFiles vector
+        UnorderedMap<core::FileRef, size_t> changedFiles;
+
+        // The names of all symbols changed by this set of updates
+        std::vector<core::ShortNameHash> changedSymbolNameHashes;
+
+        // Extra files that need to be typechecked because the file mentions the name of one of the changed symbols.
+        std::vector<core::FileRef> extraFiles;
+    };
+
+    // TODO(jez) It would be nice to have this accept `...<const core::File>...` but I can't figure
+    // out how to get the callers, which do not have `const core::File`'s. I don't know how to have
+    // C++ let me do upcasts on constness inside templates like that.
```

Comment:
You can't do this in general without copying, because `Klass<T>` and `Klass<const T>` are different types.  (`Klass<T>` and `Klass<const T>` could also have completely different representations under the hood thanks to template specialization.)

Reply:
:sadpanda: